tests/helpers/perf_run_helper.rb

```diff
@@ -944,75 +945,253 @@ def get_process_hash(perf_result_processes)
+  def atop_results_from_dir(dir, runtype)
+    # This is a dirty, dirty reconstitution of the data from a CSV file
+    # created by [beaker-benchmark](https://github.com/puppetlabs/beaker-benchmark/blob/master/lib/beaker-benchmark/helpers.rb#L219-L231).
+    file = find_atop_log_from_dir(dir, runtype)
+    csv_string = read_file(file)
+    atop_hashes = result_hashes_from_atop_csv(csv_string)
+
+    # Create PerformanceResult object
+    perf_results_args = {
+      cpu: [],
+      mem: [],
+      disk_read: [],
+      disk_write: [],
+      action: atop_hashes[:global]["Action"],
+      duration: atop_hashes[:global]["Duration"].to_f.round(2),
+      processes: {},
+      logger: nil,
+      hostname: Pathname(file).parent.split[1].to_s
+    }
+    atop_res = Beaker::DSL::BeakerBenchmark::Helpers::PerformanceResult.new(perf_results_args)
+    atop_res.avg_cpu = atop_hashes[:global]["Avg CPU"].to_i
+    atop_res.avg_mem = atop_hashes[:global]["Avg MEM"].to_i
+    atop_res.avg_disk_read = atop_hashes[:global]["Avg DSK read"].to_i
+    atop_res.avg_disk_write = atop_hashes[:global]["Avg DSK Write"].to_i
+    atop_res.processes = atop_hashes[:processes]
+    atop_res
+  end
+
+  # @param  [String] dir  Path to look for file in
+  # @param  [String] pat  Regular expression pattern to find file with
+  #
+  # @return [String]  Path to file
+  def find_file(dir, pat)
+    file = `find #{dir} -name "#{pat}" -print`.chomp
+    raise System.StandardError "The file does not exist" unless File.exist?(file)
+
+    file
+  end
+
+  # @param  [String] dir  Path where gating assertions are stored
+  #
+  # @return [String]  Path to assertions file
+  def find_gatling_assertions_from_dir(dir)
+    find_file(dir, "assertions.json")
+  end
+
+  # @param  [String] file  File to read from
+  #
+  # @return [String]  Contents of file
+  def read_file(file)
+    File.read(file)
+  end
+
+  # @param  [String] dir  Path where results are stored
+  #
+  # @return [Array<Hash>]  Array of gatling assertions hashes
+  def gatling_assertions_from_dir(dir)
+    file = find_gatling_assertions_from_dir(dir)
+    json = JSON.parse(read_file(file))
+    gatling_assertions = []
+    json["assertions"].each do |assertion|
+      gatling_assertions << { "expected_values" => assertion["expectedValues"],
+                              "message"         => assertion["message"],
+                              "actual_value"    => assertion["actualValue"],
+                              "target"          => assertion["target"] }
+    end
+    gatling_assertions
   end
-  # rubocop: enable Naming/AccessorMethodName
-end
 
+  # @param  [String] dir  Path where results are stored
+  #
+  # @return [GatlingResult]
+  def gatling_result_from_dir(dir)
+    gatling_assertions = gatling_assertions_from_dir(dir)
+    mean_response_time = mean_response_time_from_dir(dir)
+    GatlingResult.new(gatling_assertions, mean_response_time)
+  end
+
+  def find_gatling_stats_from_dir(dir)
```

Comment:
the method name is only 6 chars shorter than the code it runs... why make a method for that?  Seems like more often we define a constant with the file name at the top or something.  Not saying that is the right method... but if you think this is the right way, I would like to discuss it in the next team meeting so we can all agree to move toward doing this in the future... or not... I am a fan of consistency

Reply:
This is the result of refactoring multiple regex lookups.  My solution was to create a generic `find_file` method and then use it to abstract methods for finding desired files by there intention.  I am not married to this implementation, but it does encapsulate these lookups.  If the file used to store gatling stats changes in the future, only the pattern within this method would need to be updated.
This was the outcome of driving the code from the spec tests.  These methods made it much easier for me to control the data that I wanted to pull in to test the larger validation methods.